### PR TITLE
Add some tests for hosted project building and publishing

### PR DESCRIPTION
**Pull request by pranavkm**

### CODE REVIEW

File: src/Components/Blazor/Build/test/BuildIntegrationTests/PublishIntegrationTest.cs [hunk 1]
```diff
@@ -73,5 +103,68 @@ public async Task Publish_WithLinkOnBuildDisabled_Works()
...
+            // Make sure the main project exists
+            Assert.FileExists(result, publishDirectory, "blazorhosted.dll");
+
+            var blazorPublishDirectory = Path.Combine(publishDirectory, "standalone");
+            Assert.FileExists(result, blazorPublishDirectory, "dist", "_framework", "blazor.boot.json");
+            Assert.FileExists(result, blazorPublishDirectory, "dist", "_framework", "blazor.webassembly.js");
+            Assert.FileExists(result, blazorPublishDirectory, "dist", "_framework", "wasm", "mono.wasm");
+            Assert.FileExists(result, blazorPublishDirectory, "dist", "_framework", "wasm", "mono.js");
+            Assert.FileExists(result, blazorPublishDirectory, "dist", "_framework", "_bin", "standalone.dll");
+            Assert.FileExists(result, blazorPublishDirectory, "dist", "_framework", "_bin", "Microsoft.Extensions.Logging.Abstractions.dll"); // Verify dependencies are part of the output.
```
Comment (javiercn) [inline src/Components/Blazor/Build/test/BuildIntegrationTests/PublishIntegrationTest.cs]: Can you also check for the static web assets?

Comment (pranavkm) [reply src/Components/Blazor/Build/test/BuildIntegrationTests/PublishIntegrationTest.cs]: Done

**Comment by pranavkm** [comment]
@aspnet-hello triage https://dev.azure.com/dnceng/public/_build/results?buildId=440144&view=ms.vss-test-web.build-test-results-tab&runId=14009778&resultId=100459&paneView=debug

**Comment by aspnet-hello** [comment]
This comment was made automatically. If there is a problem contact aspnetcore-build@microsoft.com.

I've triaged the above build. I've created/commented on the following issue(s)
https://github.com/aspnet/AspNetCore-Internal/issues/3323
https://github.com/aspnet/AspNetCore-Internal/issues/2181

**Review by javiercn — APPROVED** [review]

**Pull request merged**